[lldb] Show target.debug-file-search-paths setting from  python SBDebugger

da-viper opened the pull request:
Fixes #110756

da-viper commented:
For the test I wanted to confirm this goes into the the path `/path/to/llvmproject/lldb/test/API/python_api/debugger/TestDebuggerAPI.py`

JDevlieghere commented on lldb/include/lldb/Utility/FileSpec.h:
```diff
@@ -214,6 +215,17 @@ class FileSpec {
   ///     The stream to which to dump the object description.
   void Dump(llvm::raw_ostream &s) const;
 
+  ///
```
```suggestion
```

JDevlieghere commented on lldb/include/lldb/Utility/FileSpec.h:
```diff
@@ -413,11 +424,7 @@ class FileSpec {
-    Yes,
-    No
-  };
+  enum class Absolute : uint8_t { Calculate, Yes, No };
```
Please un-stage unrelated formatting changes.

JDevlieghere commented on lldb/source/Interpreter/OptionValueFileSpecList.cpp:
```diff
@@ -41,6 +41,16 @@ void OptionValueFileSpecList::DumpValue(const ExecutionContext *exe_ctx,
+OptionValueFileSpecList::ToJSON(const ExecutionContext *exe_ctx) {
+  std::lock_guard<std::recursive_mutex> lock(m_mutex);
+  llvm::json::Array spec_list;
+  for (const auto &file_spec : m_current_value) {
+    spec_list.emplace_back(file_spec.ToJSON());
+  }
```
No braces around single-line ifs. 
```suggestion
  for (const auto &file_spec : m_current_value) 
    spec_list.emplace_back(file_spec.ToJSON());
```

JDevlieghere commented on lldb/source/Interpreter/OptionValueFileSpecList.cpp:
```diff
@@ -41,6 +41,16 @@ void OptionValueFileSpecList::DumpValue(const ExecutionContext *exe_ctx,
+llvm::json::Value
+OptionValueFileSpecList::ToJSON(const ExecutionContext *exe_ctx) {
+  std::lock_guard<std::recursive_mutex> lock(m_mutex);
+  llvm::json::Array spec_list;
```
How about `array`, that seems more appropriate? 
```suggestion
  llvm::json::Array array;
```

JDevlieghere commented on lldb/source/Interpreter/OptionValueFileSpecList.cpp:
```diff
@@ -41,6 +41,16 @@ void OptionValueFileSpecList::DumpValue(const ExecutionContext *exe_ctx,
 }
 
+llvm::json::Value
+OptionValueFileSpecList::ToJSON(const ExecutionContext *exe_ctx) {
```
It doesn't look like this actually needs the execution context?
```suggestion
OptionValueFileSpecList::ToJSON() {
```

da-viper replied:
it is  overridden from the `OptionValue` interface class

JDevlieghere commented on lldb/source/Utility/FileSpec.cpp:
```diff
@@ -118,53 +118,55 @@ bool needsNormalization(const llvm::StringRef &path) {
   for (auto i = path.find_first_of("\\/"); i != llvm::StringRef::npos;
        i = path.find_first_of("\\/", i + 1)) {
-    const auto next = safeCharAtIndex(path, i+1);
+    const auto next = safeCharAtIndex(path, i + 1);
```
I think these are all unrelated changes? Please use `git clang-format` to only format the lines changed in the PR.

JDevlieghere commented on lldb/source/Utility/FileSpec.cpp:
```diff
@@ -330,6 +332,13 @@ void FileSpec::Dump(llvm::raw_ostream &s) const {
 }
 
+llvm::json::Value FileSpec::ToJSON() const {
+  std::string file_spec{};
```
The name here is confusing, if I see `file_spec` I expect it to be a `FileSpec`, but it's a `std::string`. 

```suggestion
  std::string str{};
```

JDevlieghere commented on lldb/source/Utility/FileSpec.cpp:
```diff
@@ -330,6 +332,13 @@ void FileSpec::Dump(llvm::raw_ostream &s) const {
 
+llvm::json::Value FileSpec::ToJSON() const {
+  std::string file_spec{};
+  llvm::raw_string_ostream stream(file_spec);
+  this->Dump(stream);
+  return llvm::json::Value(std::move(file_spec));
```
```suggestion
  return llvm::json::Value(file_spec.GetPath());
```

da-viper replied:
had to move it since  json::Value accept a copy of std::string.

da-viper committed:
[lldb] Add JSON serialization for FileSpec and FileSpecList

Add a `ToJSON` method in FileSpec and OptionValueFileSpecList to enable JSON serialization.

da-viper committed:
Update lldb/include/lldb/Utility/FileSpec.h

Co-authored-by: Jonas Devlieghere <jonas@devlieghere.com>

da-viper committed:
Update lldb/source/Interpreter/OptionValueFileSpecList.cpp

Co-authored-by: Jonas Devlieghere <jonas@devlieghere.com>

da-viper committed:
[lldb] update requested changes.

JDevlieghere commented on lldb/source/Utility/FileSpec.cpp:
```diff
@@ -330,6 +330,13 @@ void FileSpec::Dump(llvm::raw_ostream &s) const {
 }
 
+llvm::json::Value FileSpec::ToJSON() const {
+  std::string str{};
```
```suggestion
  std::string str;
```

JDevlieghere commented:
This will need a test.

da-viper committed:
[lldb] Add test for OptionValueFileSpec and OptionValueFileSpecList

Signed-off-by: Ebuka Ezike <yerimyah1@gmail.com>

slydiman commented:
Now https://lab.llvm.org/buildbot/#/builders/162/builds/18611 and https://lab.llvm.org/buildbot/#/builders/195/builds/6519 are broken too. Please fix them ASAP.
